docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,459 @@
+4) ``\delta_{l}`` < `threshold` and ``\delta_{t}`` < `threshold` : In this case, we merge the top and left semgents together and assign the point under consideration to this merged segment.
+
+This algorithm segments the image in just two passes (one for segmenting and other for
+merging), hence it is very fast and can be used in real time applications.
+
+**Time Complexity:** ``O(n)`` where ``n`` is the number of pixels
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, TestImages;
+julia> img = testimage("camera");
+julia> seg = fast_scanning(img, 0.1);  # threshold = 0.1
+julia> seg = prune_segments(seg, i->(segment_pixel_count(seg,i)<50), (i,j)->(-segment_pixel_count(seg,j)))
+```
+
+**Original:**
+
+![Original](assets/segmentation/camera.jpg)
+
+**Segmented Image:**
+
+![SegmentedImage](assets/segmentation/camera_seg.jpg)
+
+#### Region Splitting using RegionTrees
+
+This algorithm follows the divide and conquer methodology. If the input
+image is homogeneous then nothing is to be done. In the other case, the
+image is split into two across every dimension and the smaller parts are
+segmented recursively. This procedure generates a region tree which can
+be used to create a segmented image.
+
+**Time Complexity:** ``O(n*log(n))`` where ``n`` is the number of pixels
+
+###### Demo
+
+```julia
+julia> using TestImages, ImageSegmentation;
+julia> img = testimage("lena_gray")
+julia> function homogeneous(img)
+           min, max = extrema(img)
+           max - min < 0.2
+       end
+julia> seg = region_splitting(img, homogeneous);
+```
+
+**Original:**
+
+![Original](assets/segmentation/lena.jpg)
+
+**Segmented Image with labels replaced by their intensity means:**
+
+![SegmentedImage](assets/segmentation/lena_seg.jpg)
+
+#### Fuzzy C-means
+
+Fuzzy C-means clustering is widely used for unsupervised image segmentation. It is an
+iterative algorithm which tries to minimize the cost function:
+
+```math
+J = \displaystyle\sum_{i=1}^{N} \sum_{j=1}^{C} \delta_{ij} \| x_{i} - c_{j} \|^2
+```
+
+Unlike K-means, it allows pixels to belong to two or more clusters. It is widely used
+for medical imaging like in the soft segmentation of brain tissue model.
+
+**Time Complexity:** ``O(n*C^2*iter)`` where ``n`` is the number of pixels, ``C`` is
+number of clusters and ``iter`` is the number of iterations.
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, Images;
+julia> img = load("flower.jpg");
+julia> r = fuzzy_cmeans(img, 3, 2);
+```
+
+**Original** [(source)](https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Flower-631765_960_720.jpg/800px-Flower-631765_960_720.jpg)
+
+
+![Original](assets/segmentation/flower.jpg)
+
+**Output with pixel intensity = cluster center intensity * membership of pixel in that class**
+
+![SegmentedImage1](assets/segmentation/flower_s1.jpg) ![SegmentedImage2](assets/segmentation/flower_s2.jpg)
+![SegmentedImage3](assets/segmentation/flower_s3.jpg)
+
```

Comment:
A few verbal comments might be in order. What do the 3 clusters "mean"? (Obvious to you, but worth explaining that cluster 1 corresponds to the magenta petals, cluster 2 to the leaves, and cluster 3 to the white background.) Also, out of curiosity why doesn't the "leaves cluster" look green?

Reply:
The "leaves cluster" includes pixels other than green that are close to the cluster center in the RGB space. Also the "image pixel intensity = center intensity * membership" rather than just the center intensity, so color is kind of greenish-black.
![greenish](https://user-images.githubusercontent.com/15063205/29740714-57f6d1d8-8a7a-11e7-988c-337371eb50b9.jpg)
In the above image, each pixel intensity denotes the membership of the pixel in the "leaves cluster". As can be seen, many pixels that are not green also contribute fairly to this cluster.